### ISSUE TITLE
"Skip ad" button not clickable

### ISSUE THREAD

**Issue by dave1981** [google/shaka-player#3284]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
3.0.10

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `master`?**
I don't know

**Are you using the demo app or your own custom app?**
custom app

**If custom app, can you reproduce the issue using our demo app?**
No

**What browser and OS are you using?**
Windows 10 - Firefox latest version

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**

**What are the manifest and license server URIs?**

**What configuration are you using?  What is the output of `player.getConfiguration()`?**

**What did you do?**
I'm trying to build a Video Player with ads, at the moment only before playing. I'm working with Angular, I included shaka-player.ui.js, and I followed the guides (https://shaka-player-demo.appspot.com/docs/api/tutorial-ui.html and https://shaka-player-demo.appspot.com/docs/api/tutorial-ad_monetization.html) to setup the ui via typescript and to integrate client side ads.
My server returns an XML with a sample video and the possibility to skip the ad after 3 seconds.

**What did you expect to happen?**
What I expect is that, when I click the button "skip ad", the ad is closed and the video starts playing.

**What actually happened?**
The button "skip ad" seems to be unclickable.

**Comment by ismena** [comment]
@dave1981 Interesting. Does it work if you try the same manifest + same ads on our demo page?
(Asking to disambig between a Shaka bug and a Shaka+angular integration problem)

**Comment by dave1981** [comment]
No, there's the same problem using

Manifest URL (from https://gist.github.com/jsturgis/3b19447b304616f18657): 
http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4

Ad tag URL (from https://foliovision.com/player/vast/sample-vast-tags):
https://pubads.g.doubleclick.net/gampad/ads?sz=640x480&iu=/124319096/external/single_ad_samples&ciu_szs=300x250&impl=s&gdfp_req=1&env=vp&output=vast&unviewed_position_start=1&cust_params=deployment%3Ddevsite%26sample_ct%3Dskippablelinear&correlator=1616744579

**Comment by dezfuture** [comment]
I would like to contribute and fix the issue.  Can you guide me on how should I go forward into solving this issue

**Comment by ismena** [comment]
@dave1981 what an awesome report :) thanks! Should be easy enough to fix.

@dezfuture You're welcome to contribute. Here is how it should work and what we do:
The "skip ad" button comes from the ad SDK, it's not ours. It lives in an iframe that we lift up and increase its z-index [here](https://github.com/google/shaka-player/blob/master/ui/less/ad_controls.less#L40). 
So, 1) it doesn't look to be high enough - the middle of the button should be inline with the "Ad X of Y" thing.
and 2) looks like the z-index setting isn't taking effect - the i-frame should be "in front" of the ad-controls div.

Once you have a fix, please check all major browsers, play around with shrinking the screen, and verify the fullscreen mode.
Thanks!

**Comment by dave1981** [comment]
@ismena a little question: when I start playing a video with ads, I see a message "Shaka Player's ad features are currently in BETA and are NOT yet covered by semantic versioning compatibility guarantees.  The API may change at any time!". Do you have a roadmap to release this feature ready for production use?

**Comment by meryacine** [comment]
@ismena I think this part 
```
/* IMA SDK adds their own ad UI into an iframe element.
   * Adjust its position to fit in with our UI, when
   * Shaka UI is enabled. */
  iframe {
    .shaka-video-container[shaka-controls="true"] & {
      /* This moves the iframe up a little bit, so it
       * doesn't operlap with our controls. */
      height: 92%;
      z-index: 1;
    }
```
should also be here https://github.com/google/shaka-player/blob/b4595d5a594f64e813a7d40f8e1a1bc77a3589dd/ui/less/ad_controls.less#L32 based on this commit https://github.com/google/shaka-player/commit/655493f441f8840ac8f4b9f3053323bfbd1292f7.

Doing so leaves the SkipAd button too high .

![Screenshot1](https://user-images.githubusercontent.com/56920956/113687200-5e9e7280-96c8-11eb-8664-69c17b8ed9d6.png)
In this screen shot i used ```height:80%``` because ```92%``` seems to still interfere with the controls.
<hr>

![Screenshot2](https://user-images.githubusercontent.com/56920956/113688192-5b57b680-96c9-11eb-9e8d-727d53ffbb1e.png)
The SkipAd button after applying ```bottom:0px``` & ```padding-bottom:0px``` to its container.

So the question is, can i make a CSS selection that selects the container of the SkipAd button and applies the ```bottom:0px``` & ```padding-bottom:0px``` to it in order to bring the button further down like in the second screen shot, or this would be a bad thing to do (editing the UI coming from IMA).
Any other idea ?

**Comment by ismena** [comment]
@meryacine Ah, good point about the structure having changed in 655493f. Let me assign it to myself for investigation. As there's community interest on working on this, I won't fix anything and will just update the issue with the best course of action.
I'll try to look at this next week.

BTW, what browser did you use for the screenshots? The volume slider's thumb looks messed up :(

**Comment by meryacine** [comment]
Oke :D ,
the browser was chrome, but firefox does the same, this happened when i used lessc[v4.1.1] to compile the controls.less instead of lessc[v3.13.1] used by build/app.py

**Comment by ismena** [comment]
Okay! So, it looks like now that the IMA iframe is inside the `shaka-client-side-ad-container`, [this code](https://github.com/google/shaka-player/blob/master/ui/less/ad_controls.less#L40) needs to move from the `controls-container` into the `client-side-container`.

It also seems like we no longer need the z-index setting.

**Comment by meryacine** [comment]
can u also look at this issue for server-side-ads
![Screenshot from 2021-04-09 20-52-35](https://user-images.githubusercontent.com/56920956/114227792-bd3c4880-9975-11eb-9db2-cfe0d2f3f553.png)
caused by inherited ```flex-shrink: 0```, can be fixed using ```.shrinkable();``` here https://github.com/google/shaka-player/blob/2f65b97b79c9b9dfde1b95afadf7c790827054c7/ui/less/ad_controls.less#L24

**Comment by ismena** [comment]
Yeah, you'll need to string server side and client side containers together in the less file, so it's applied to both.
Re: button being too high - I can live with the way it is, but you can also adjust the percent value applied to the iframe. If you do, verify that it works in fullscreen and in all major browsers.

You can go ahead with the` .shrinkable()` fix, that LGTM.

**Comment by ismena** [comment]
Fixed by  78e181d
